src/snowflake/cli/api/utils/definition_rendering.py

```diff
@@ -0,0 +1,191 @@
+from __future__ import annotations
+
+import copy
+import os
+from dataclasses import dataclass
+from typing import List
+
+import yaml
+from jinja2 import Environment, UndefinedError, nodes
+from snowflake.cli.api.project.schemas.project_definition import ProjectDefinition
+from snowflake.cli.api.utils.rendering import get_snowflake_cli_jinja_env
+
+
+@dataclass
+class GraphNode:
+    path: str
+    value: str | None = None
+
+
+def _get_referenced_vars(ast_node, attr_chain: List[str] = []) -> List[List[str]]:
```

Comment:
I'm having trouble visualizing what this type represents (`List[List[str]]`). Can you provide a type alias?